compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1730,7 +1730,7 @@ class Typer extends Namer
   def ensureConstrCall(cls: ClassSymbol, parents: List[Tree])(implicit ctx: Context): List[Tree] = {
     val firstParent :: otherParents = parents
     if (firstParent.isType && !(cls is Trait) && !cls.is(JavaDefined))
-      typed(untpd.New(untpd.TypedSplice(firstParent), Nil)) :: otherParents
+      typed(untpd.New(untpd.TypedSplice(firstParent).withSpan(firstParent.span), Nil)) :: otherParents
```

Comment:
Hmm, why isn't the span of TypedSplice automatically inferred from the span of its only child ?

Reply:
Not sure. I will loo into it.

Reply:
It looks like a possibility. I will implement it.

Reply:
That looks indeed redundant.